### ISSUE TITLE
Port registration networks

### ISSUE THREAD

**Issue by wyli** [Project-MONAI/MONAI#95]
a subtask of #78, port registration networks

ref:
https://github.com/YipengHu/label-reg
https://niftynet.readthedocs.io/en/dev/niftynet.network.interventional_hybrid_net.html

@yipenghu

**Comment by tvercaut** [comment]
Not sure if this fits in this issue but ideally, for this purpose, MONAI would provide efficient (i.e. on the GPU) means of performing image resampling. For NiftyNet, this was somewhat solved by embedding resampling code from NiftyReg:
https://niftynet.readthedocs.io/en/dev/niftynet.contrib.niftyreg_image_resampling.html

On the long run a more pytorch friendly option might be relevant.

On a related note, it's worth mentioning that DALI provide 2D/3D resampling as well but it seems restricted to affine warps:
https://docs.nvidia.com/deeplearning/sdk/dali-developer-guide/docs/examples/warp_3D.html

**Comment by tvercaut** [comment]
@mjorgecardoso rightly pointed out that pytorch provides a very handy `grid-sample` function which features a GPU implementation: https://pytorch.org/docs/stable/nn.functional.html#grid-sample

**Comment by tvercaut** [comment]
As per our meeting, these efforts are of interest:
https://github.com/DeepRegNet/DeepReg
https://github.com/airlab-unibas/airlab

Also https://github.com/voxelmorph/voxelmorph/tree/master/pytorch but note that the code there is GPL licensed.

**Comment by ciphercharly** [comment]
i am not sure this is the right place to ask, but i need to implement a 3d registration network for registering MRI image, and i am looking at projects like DeepReg, EasyReg, label-reg, quicksilver.. and i was wondering which project is the most likely to be ported in MONAI? future compatibility with MONAI would be good to have, so i wanna choose wisely where to start..

**Comment by YipengHu** [comment]
> i am not sure this is the right place to ask, but i need to implement a 3d registration network for registering MRI image, and i am looking at projects like DeepReg, EasyReg, label-reg, quicksilver.. and i was wondering which project is the most likely to be ported in MONAI? future compatibility with MONAI would be good to have, so i wanna choose wisely where to start..

I can speak for label-reg, which has been discontinued. All its functionalities and the superset have been implemented in DeepReg with newer tf2.

**Comment by ciphercharly** [comment]
good to know, thanks

**Comment by Spenhouet** [comment]
Does this issue mean the implementation of DeepReg with Pytorch in MONAI? Not sure from the comments.

Is anyone on this?

**Comment by wyli** [comment]
closing this in favour of #1651 #1665. see also #1658
